app/controllers/contacts.js

```diff
@@ -0,0 +1,109 @@
+import BaseController from "./base/base.js";
+import Contact from "./../models/contact.js";
+import MatchedRepository from "../repositories/matched_repository.js";
+import SessionRepository from "./../repositories/session_repository.js";
+import { ACTIVE } from "./../store/session.js";
+import { ERROR_STATUES } from "./../validations/constants/errors.js";
+import { ObjectId } from "mongodb";
+
+class ContactsController extends BaseController {
+  constructor() {
+    super();
+
+    this.sessionRepository = new SessionRepository(ACTIVE);
+    this.matchedRepository = new MatchedRepository();
+  }
+
+  async contact_add(ws, data) {
+    const { id: requestId, contact_add: contactData } = data;
+    const currentUser = this.sessionRepository.getSessionUserId(ws);
+
+    contactData.email &&
+      (await this.matchedRepository.matchedUser(contactData, "email"));
```

Comment:
hmm, I see we pass here a diff object as 1st param.
From users controller we pass `user.visibleParams()` and from here we pass contact.
I think it can produce lots of unclearity. Can we improve it?

Reply:
also, here we pass `"email"` as options, but from users controller we pass an object, not a string. 

If we want to pass 2 diff data formats, let's have 2 diff methods for it